File: alwin-middleware-grapescode/alwin-frontend-jpa/src/main/java/com/codersteam/alwin/jpa/TagWallet.java
package com.codersteam.alwin.jpa;

import com.codersteam.alwin.jpa.issue.Tag;

import java.math.BigDecimal;

/**
 * Portfele zleceń liczone dla zleceń przypisanych do etykiety.
 * Portfele są liczone z otwartych zleceń windykacyjnych.
 *
 * @author Piotr Naroznik
 */
public class TagWallet extends Wallet {

    /**
     * Etykieta dla której liczony jest portfel
     */
    private final Tag tag;

    public TagWallet(final Long companyCount, final BigDecimal involvement, final BigDecimal balanceStartPLN,
                     final BigDecimal balanceStartEUR, final Tag tag) {
        super(companyCount, involvement, balanceStartPLN, balanceStartEUR);
        this.tag = tag;
    }

    public Tag getTag() {
        return tag;
    }

    @Override
    public String toString() {
        return "TagWallet{" +
                "tag=" + tag +
                "} " + super.toString();
    }
}

File: alwin-web-grapescode/src/app/segment/tag-wallet.ts
import {Wallet} from './wallet';
import {Tag} from '../tag/tag';

/**
 * Portfele zleceń liczone dla zleceń przypisanych do etykiety.
 * Portfele są liczone z otwartych zleceń windykacyjnych.
 */
export class TagWallet extends Wallet {

  /**
   * Etykieta dla której liczony jest portfel
   */
  tag: Tag;
}
